vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/EntityTest.java

```diff
@@ -0,0 +1,152 @@
+import org.semanticweb.vlog4j.core.model.implementation.FactImpl;
+import org.semanticweb.vlog4j.core.model.implementation.LanguageStringConstantImpl;
+import org.semanticweb.vlog4j.core.model.implementation.PredicateImpl;
+import org.semanticweb.vlog4j.core.model.implementation.RuleImpl;
+import org.semanticweb.vlog4j.core.reasoner.KnowledgeBase;
+import org.semanticweb.vlog4j.core.reasoner.implementation.CsvFileDataSource;
+import org.semanticweb.vlog4j.core.reasoner.*;
+import org.semanticweb.vlog4j.core.reasoner.implementation.RdfFileDataSource;
+import org.semanticweb.vlog4j.core.reasoner.implementation.SparqlQueryResultDataSource;
+import org.semanticweb.vlog4j.parser.ParsingException;
+import org.semanticweb.vlog4j.parser.RuleParser;
+
+public class EntityTest {
+	final Variable x = Expressions.makeUniversalVariable("X");
+	final Variable y = Expressions.makeUniversalVariable("Y");
+	final Variable z = Expressions.makeExistentialVariable("Z");
+	final Variable y2 = Expressions.makeUniversalVariable("Y");
+	final Constant d = Expressions.makeAbstractConstant("d");
+	final Constant c = Expressions.makeAbstractConstant("c");
+	final AbstractConstantImpl f = new AbstractConstantImpl("f");
+	final LanguageStringConstantImpl s = new LanguageStringConstantImpl("Test", "en");
+	final DatatypeConstantImpl data = new DatatypeConstantImpl("data", "http://example.org/mystring");
+	final PositiveLiteral atom1 = Expressions.makePositiveLiteral("p", x, c);
+	final PositiveLiteral atom2 = Expressions.makePositiveLiteral("p", x, y);
+	final PositiveLiteral headAtom1 = Expressions.makePositiveLiteral("q", x, z);
+	final PositiveLiteral positiveLiteral1 = Expressions.makePositiveLiteral("p", x, c);
+	final PositiveLiteral positiveLiteral2 = Expressions.makePositiveLiteral("p", y2, x);
+	final PositiveLiteral positiveLiteral3 = Expressions.makePositiveLiteral("q", x, d);
+	final NegativeLiteral NegativeLiteral = Expressions.makeNegativeLiteral("r", x, d);
+	final PositiveLiteral PositiveLiteral4 = Expressions.makePositiveLiteral("s", c, s);
+	final Predicate p = Expressions.makePredicate("p", 2);
+	final Fact f1 = Expressions.makeFact(p, Arrays.asList(f, s));
+	final Fact f2 = Expressions.makeFact("p", Arrays.asList(data, d));
+	final List<Literal> LiteralList = Arrays.asList(positiveLiteral1, positiveLiteral2, positiveLiteral3,
+			NegativeLiteral, PositiveLiteral4);
+	final Conjunction<Literal> bodyLiterals = Expressions.makeConjunction(atom1, atom2);
+	final Conjunction<PositiveLiteral> headPositiveLiterals = Expressions.makePositiveConjunction(headAtom1);
+	final Conjunction<Literal> bodyConjunction = new ConjunctionImpl<>(LiteralList);
+	final Rule rule1 = new RuleImpl(headPositiveLiterals, bodyLiterals);
+	final Rule rule2 = new RuleImpl(headPositiveLiterals, bodyConjunction);
+
+	@Test
+	public void factToStringRoundTripTest() {
+		KnowledgeBase kb = new KnowledgeBase();
+		KnowledgeBase kb2 = new KnowledgeBase();
+		kb.addStatement(f1);
```

Comment:
why do you add things to the knowledge base?

Reply:
because I don't want to give a certain string I wanted to create the object then parse and then I can check whether the string is the same as the one before the parsing